### DISCUSSION TITLE
FileStream rewrite?

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#49783] [General] [answered]
It looks like the FileStream class is getting a lot of love. It would be great if there was a blog post about this. I'm really curious about the before and after differences.

**Comment by stephentoub** [comment, marked as answer]
A blog post is part of @adamsitnik's plan.

**Comment by adamsitnik** [comment]
@TonyValenti The blog post is out: https://devblogs.microsoft.com/dotnet/file-io-improvements-in-dotnet-6/

**Comment by TonyValenti** [comment]
@adamsitnik Wowzers!  What a great post!  Thanks so much for putting in all the time necessary to write it.  Also, I would be interested in reading ```.NET File IO performance guidelines``` if you write it.

A question I have:
Is there any room for ```StreamReader``` and ```StreamWriter``` to leverage the performance enhancements that have been made to FileStream?  I'm specifically thinking around the buffer size stuff.

**Comment by adamsitnik** [comment]
> Is there any room for StreamReader and StreamWriter to leverage the performance enhancements that have been made to FileStream?

It depends on usage patterns. All `*Async` methods should definitely get a very nice perf boost.
